### PR TITLE
Build improvements

### PR DESCRIPTION
**Pull request by johnhaddon**
This :

- Declares dependencies between projects and uses them to automatically build projects in order, and build subsets of projects with their required dependencies.
- Keeps track of a build digest to avoid building things that haven't changed.
- Moves all build logic into `./build.py`, so we no longer have a mix of shell and python hidden away in a subdirectory.

### CODE REVIEW

File: build/build.py [hunk 1]
```diff
@@ -11,6 +11,19 @@
...
+__version = "0.56.0.0"
+
+def __globalVariables( buildDir ) :
+
+	return {
+		"buildDir" : buildDir,
+		"jobs" : multiprocessing.cpu_count(),
+		"path" : os.environ["PATH"],
+		"version" : __version,
+		"platform" : "osx" if sys.platform == "darwin" else "linux",
```
Comment (themissingcow) [inline build/build.py]: Can we make this windows friendly to save Eric having to come back to it?

Comment (themissingcow) [inline build/build.py]: We need to remember to move to `macos` at some point when we're ready to update `gaffer` too.

Comment (johnhaddon) [reply build/build.py]: There's a lot more work than this to make this project windows-friendly, and although it's a trivial change I'm sure, I don't have any way of testing...

Comment (johnhaddon) [reply build/build.py]: What does the update in gaffer entail? Seems like maybe I should change it now?

Comment (themissingcow) [reply build/build.py]: I think there is:
https://github.com/GafferHQ/gaffer/blob/3d42aa2051f2a07932b02724ccc7b138921b5d45/config/installDependencies.sh#L50
and:
https://github.com/GafferHQ/build/blob/cb9e4594879a1f49f551bb1ef1061d932882638a/build.py#L159
May be others

Comment (johnhaddon) [reply build/build.py]: Well, this is going to be gafferDependencies-1, and we'll need to change Gaffer to pick it up anyway, so how about I just do it now?

Comment (themissingcow) [reply build/build.py]: Sounds good to me!

**Pull request merged**